src/spork.cpp

```diff
@@ -347,6 +408,30 @@ bool CSporkMessage::CheckSignature(const CKeyID& pubKeyId, bool fSporkSixActive)
     return true;
 }
 
+bool CSporkMessage::GetSignerKeyID(CKeyID &sporkSignerID, bool fSporkSixActive)
+{
+    CPubKey pubkeyFromSig;
+    if (fSporkSixActive) {
+        if (!pubkeyFromSig.RecoverCompact(GetHash(), vchSig)) {
```

Comment:
Hash should include signature for this functionality because of the fact that signature is the only thing that is different in messages from different signers. If we don't include the signature the messages  from other signers are not broadcasted after the message from the first signer. So it looks like I should add some update code compatible with the old version.(

Reply:
I know :) But this is only true for p2p part, not for the sign/verify part which can use whatever hash it needs, that's basically the purpose of `GetSignatureHash()`.